kubernetes/kubernetes_service.go

```diff
@@ -84,60 +93,107 @@ func (in *IstioClient) GetDeployments(namespaceName string) (*v1beta1.Deployment
 	return in.k8s.AppsV1beta1().Deployments(namespaceName).List(emptyListOptions)
 }
 
+// GetDeployments returns a list of deployments for a given namespace.
+// It returns an error on any problem.
+func (in *IstioClient) GetService(namespaceName, serviceName string) (*v1.Service, error) {
+	return in.k8s.CoreV1().Services(namespaceName).Get(serviceName, emptyGetOptions)
+}
+
 // GetServiceDetails returns full details for a given service, consisting on service description, endpoints and pods.
 // A service is defined by the namespace and the service name.
 // It returns an error on any problem.
-func (in *IstioClient) GetServiceDetails(namespace string, serviceName string) (*ServiceDetails, error) {
-	serviceChan := make(chan serviceResponse)
+func (in *IstioClient) GetServiceDetails(namespaceName string, serviceName string) (*ServiceDetails, error) {
```

Comment:
Minor, instead to add a namespaceName, when I looked at this code, I would short the serviceName instead.
Personally namespaceName and serviceName has being mutated in a strange way.

Reply:
Fine by me, I think namespaceName is verbose, changed for consistency but in the wrong way :)

Reply:
Point is you change this one, I guess you should change all signatures, too for consistency.
So, all or nothing :-)

Reply:
I changed it to be consistent with all of the other ones.  They were all already namespaceName.  Anyway, Edgar can update as he sees fit.

Reply:
You are right. There were a mix of signatures on that file.